tracing-opentelemetry/src/span_ext.rs

```diff
@@ -86,6 +120,26 @@ impl OpenTelemetrySpanExt for tracing::Span {
         });
     }
 
+    fn add_link(&self, cx: Context) {
+        let mut cx = Some(cx);
+        self.with_collector(move |(id, collector)| {
+            if let Some(get_context) = collector.downcast_ref::<WithContext>() {
+                get_context.with_context(collector, id, move |builder, _tracer| {
+                    if let Some(cx) = cx.take() {
+                        let follows_context = cx.span().span_context().clone();
+                        let follows_link =
+                            opentelemetry::trace::Link::new(follows_context, Vec::new());
+                        if let Some(ref mut links) = builder.links {
+                            links.push(follows_link);
+                        } else {
+                            builder.links = Some(vec![follows_link]);
```

Comment:
take it or leave it: this could also be implemented as 
```suggestion
                        builder
                            .links
                            .get_or_insert_with(|| Vec::with_capacity(1))
                            .push(follows_link);
```
but it's up to you whether this is clearer...